packages/kit/src/runtime/server/page/load_data.js

```diff
@@ -79,3 +86,42 @@ async function unwrap_promises(object) {
 
 	return unwrapped;
 }
+
+/**
+ * Check that the data can safely be serialized to JSON
+ * @param {any} value
+ * @param {string} id
+ * @param {string} path
+ */
+function check_serializability(value, id, path) {
+	const type = typeof value;
+
+	if (type === 'string' || type === 'boolean' || type === 'number' || type === 'undefined') {
+		// primitives are fine
+		return;
+	}
+
+	if (type === 'object') {
+		// nulls are fine...
+		if (!value) return;
+
+		// ...so are plain arrays...
+		if (Array.isArray(value)) {
+			value.forEach((child, i) => {
+				check_serializability(child, id, `${path}[${i}]`);
+			});
+			return;
+		}
+
+		// ...and objects
+		const tag = Object.prototype.toString.call(value);
+		if (tag === '[object Object]') {
```

Comment:
luckily i know where i can find a more robust check!

```suggestion
		// This simple check might potentially run into some weird edge cases
		// Refer to https://github.com/lodash/lodash/blob/2da024c3b4f9947a48517639de7560457cd4ec6c/isPlainObject.js?rgh-link-date=2022-07-20T12%3A48%3A07Z#L30
		// if that ever happens
		if (Object.getPrototypeOf(value) === Object.prototype) {
```

Reply:
That one is essentially "if it's not a freshly constructed plain object then no bueno", 
I think it'll miss Object.defineProperties and other stuff, but if people are going to those lengths for the returned object, I wouldn't want to accomodate that.

Reply:
Also, are we going to need these checks if #6008 is implemented?

Reply:
No, in that case we'd have to rely on `devalue` throwing an error if stuff isn't serializable. Though it would be preferable to serialize the entire array in one go, which would make it tricker to track down the error... I haven't really thought about the best way to do that yet, but in any case #6008 probably won't happen for a while (feels like a post-1.0 thing)